pyroomacoustics/libroom_src/room.cpp

```diff
@@ -693,13 +693,17 @@ bool Room<D>::scat_ray(
     {
       // As the ray is shot towards the microphone center,
       // the hop dist can be easily computed
-      float hop_dist = (hit_point - mic_pos).norm();
+      Vectorf<D> hit_point_to_mic = mic_pos - hit_point;
```

Comment:
So the vector direction is pointing towards the wall. I believe this is necessary to get the correct angle computation for the Lambert scattering, because the wall normal is pointing behind the wall in our convention. Is this correct ? If so, we could add the explanation in a comment.

Reply:
Oups! You are right! Yes, you can look for comments in the code to verify, but I think I set the default to be normals pointing outside the room.

Reply:
In fact, it might be better to wrap the angle computation in a dedicated method of the wall class so that we don't have to worry about this anymore.

Reply:
Good idea 👍 

Actually looking at [how the normal](https://github.com/LCAV/pyroomacoustics/blob/cb914279b6568635856743322175e2b3c65aed6c/pyroomacoustics/libroom_src/wall.cpp#L81) is computed I'm not sure if it is always pointing outside the room. From just two corners, how could it be known which direction is in/out? But I guess in a dedicated method, we can make the assumption that the mic is in the room in order to perform the correct angle computation?